feature/experimental/bgp/ate_tests/bgp_2byte_4byte_asn/bgp_2byte_4byte_asn_test.go

```diff
@@ -0,0 +1,257 @@
+	resetBatch := &gnmi.SetBatch{}
+	gnmi.BatchDelete(resetBatch, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Config())
+
+	if deviations.NetworkInstanceTableDeletionRequired(dut) {
+		tablePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).TableAny()
+		for _, table := range gnmi.LookupAll(t, dut, tablePath.Config()) {
+			if val, ok := table.Val(); ok {
+				if val.GetProtocol() == oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP {
+					gnmi.BatchDelete(resetBatch, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Table(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, val.GetAddressFamily()).Config())
+				}
+			}
+		}
+	}
+	resetBatch.Set(t, dut)
+}
+
+func verifyPeer(t *testing.T, nbr *bgpNbr, dut *ondatra.DUTDevice) {
+	t.Helper()
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(nbr.peerIP)
+	glblPath := statePath.Global()
+
+	// Check BGP peerAS from telemetry.
+	peerAS := gnmi.Get(t, dut, nbrPath.State()).GetPeerAs()
+	if peerAS != nbr.peerAS {
+		t.Errorf("BGP peerAs: got %v, want %v", peerAS, nbr.peerAS)
+	}
+
+	// Check BGP localAS from telemetry.
+	localAS := gnmi.Get(t, dut, nbrPath.State()).GetLocalAs()
+	if localAS != nbr.localAS {
+		t.Errorf("BGP localAS: got %v, want %v", localAS, nbr.localAS)
+	}
+
+	// Check BGP globalAS from telemetry.
+	globalAS := gnmi.Get(t, dut, glblPath.State()).GetAs()
+	if globalAS != nbr.localAS {
+		t.Errorf("BGP globalAS: got %v, want %v", globalAS, nbr.localAS)
+	}
+}
+
+func configureATE(t *testing.T, ateParams *bgpNbr, connectionType string) *ondatra.ATETopology {
+	t.Helper()
+	t.Log("Configure ATE interface")
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+
+	iDut1 := topo.AddInterface(ateSrc.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
+	iDut1.IPv6().WithAddress(ateSrc.IPv6CIDR()).WithDefaultGateway(dutSrc.IPv6)
+
+	bgpDut1 := iDut1.BGP()
+
+	peer := bgpDut1.AddPeer().WithPeerAddress(ateParams.peerIP).WithLocalASN(ateParams.localAS)
+	if connectionType == connInternal {
+		peer.WithTypeInternal()
+	} else {
+		peer.WithTypeExternal()
+	}
+	return topo
+}
+
+func createBgpNeighbor(nbr *bgpNbr, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	ni_proto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := ni_proto.GetOrCreateBgp()
+
+	global := bgp.GetOrCreateGlobal()
+	global.As = ygot.Uint32(nbr.localAS)
+	global.RouterId = ygot.String(dutSrc.IPv4)
+
+	pg := bgp.GetOrCreatePeerGroup("ATE")
+	pg.PeerAs = ygot.Uint32(nbr.peerAS)
+	pg.PeerGroupName = ygot.String("ATE")
+
+	if nbr.isV4 {
+		nv4 := bgp.GetOrCreateNeighbor(nbr.peerIP)
```

Comment:
the first four lines of the if- and else- branches are the same except for the name of the variable. Rename the variable to "neighbor" and move it before the if-statement, so that both branches share that logic